crates/celo-revm/src/common/fee_currency_context.rs

```diff
@@ -65,6 +65,18 @@ impl FeeCurrencyContext {
         }
     }
 
+    pub fn max_allowed_currency_intrinsic_gas_cost(
+        &self,
+        currency: Option<Address>,
```

Comment:
Do we ever call this with currency = None? I would remove the Option and not handle the case here. We must not call any function with the resulting 0 gas value anyway, so it is better to handle the case elsewhere.